ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md

```diff
@@ -789,6 +793,10 @@ port_channel_interfaces:
       dot1q:
         vlan: < vlan tag to configure on sub-interface >
     mode: < access | dot1q-tunnel | trunk >
+    l2_protocol:
+      encapsulation:
```

Comment:
there is only one option with `encapsulation` in EOS. That is `encapsulation dot1q vlan`. I think it is a bit much to do all these nested dicts. @aphillipps @titom73 ?

Reply:
+1 for this approach!

Reply:
Initial proposition is the strict application of data model definition. But I am fine with this one considering there is nothing else to come under encapsulation.

Reply:
ok. I can change the l2-protocol encapsulation to the new suggested model (encapsulation_dot1q_vlan), but then I will change these ones too if you all agree so that it stays consistent across AVD:

Currently
```yaml
ethernet_interfaces:
  < Ethernet_interface_1 >:
    encapsulation:
      dot1q:
        vlan: < vlan tag to configure on sub-interface >
    l2_protocol:
      encapsulation:
        dot1q:
          vlan: < vlan number >
!
port_channel_interfaces:
  < Port-Channel_interface_1 >:
    encapsulation:
      dot1q:
        vlan: < vlan tag to configure on sub-interface >
    l2_protocol:
      encapsulation:
        dot1q:
          vlan: < vlan number >
```

After change

```yaml
ethernet_interfaces:
  < Ethernet_interface_1 >:
    encapsulation_dot1q_vlan: < vlan tag to configure on sub-interface >
    l2_protocol:
      encapsulation_dot1q_vlan: < vlan number >
!
port_channel_interfaces:
  < Port-Channel_interface_1 >:
    encapsulation_dot1q_vlan: < vlan tag to configure on sub-interface >
    l2_protocol:
      encapsulation_dot1q_vlan: < vlan number >
```

Please let me know...

Reply:
Yes, I agree!
Since we will implement in 2.x.x train we would need to keep backward compatibility with the old data model until 3.x.x. release.